### PR TITLE
Search for packages that contain a given class

### PR DESCRIPTION
**Pull request by maddingo**
The new command class-search lists the packages that contain a certain class, given by its full name

### CODE REVIEW

File: src/main/java/it/mulders/mcs/cli/Cli.java [hunk 1]
```diff
@@ -45,10 +49,26 @@ public SearchCommand() {
         }
 
         @Override
-        public Integer call() throws Exception {
+        public Integer call() {
```
[the remark this answers is not in the document]
Comment (maddingo) [reply src/main/java/it/mulders/mcs/cli/Cli.java]: Done

File: src/test/java/it/mulders/mcs/search/NoOutputPrinterTest.java [hunk 2]
```diff
@@ -1,6 +1,5 @@
 package it.mulders.mcs.search;
 
-import org.apache.commons.io.output.NullOutputStream;
```
[the remark this answers is not in the document]
Comment (maddingo) [reply src/test/java/it/mulders/mcs/search/NoOutputPrinterTest.java]: Done

File: src/test/java/it/mulders/mcs/search/SearchClientIT.java [hunk 3]
```diff
@@ -102,17 +103,45 @@ void should_parse_response_groupId_artifactId_version(final WireMockRuntimeInfo
...
+            assertThat(ids).containsOnly("org.junit.jupiter:junit-jupiter-api");
+        }
+    }
+
     @DisplayName("Error handling")
     @Nested
     class ErrorHandlingTest {
         @Test
-        void should_gracefully_return_failures(final WireMockRuntimeInfo wmRuntimeInfo) throws MalformedURLException {
+        void should_gracefully_return_failures(final WireMockRuntimeInfo wmRuntimeInfo) {
```
[the remark this answers is not in the document]
Comment (maddingo) [reply src/test/java/it/mulders/mcs/search/SearchClientIT.java]: done

**Comment by maddingo** [comment]
I still feel that using a new search command is adequate.
What would  `mcs search -c LogManager log4j` do? should it do a search for both an artifact that contains log4j and a library that contains a class Logmanager. Should it be the intersection of both? 

Having two distinct search command emphasizes that these are two different searches. What do you think?

**Comment by maddingo** [comment]
I will resubmit the feature in another PR due to lot of changes in the target branch

**Comment by mthmulders** [comment]
I'm sorry for the inconvenience the refactoring may have caused. Perhaps it's easier to propose the changes on a separate branch?

I'll get back to the command/flag matter, hopefully later today.

**Comment by mthmulders** [comment]
@all-contributors please add @maddingo for code.

**Pull request closed without merging**